### ISSUE TITLE
Can't set defaults in __init__

### ISSUE THREAD

**Issue by fuzzthink** [seq-lang/seq#110]
```python
class Test:
   def __init__(self:Test, k=0):
      print k
```

test.seq:11:1: error: magic method overrides cannot be generic (__init__)

**Comment by arshajii** [comment]
Yes, this is currently a limitation, but you can overload `__init__`:

```python
class Test:
    def __init__(self: Test):
        k = 0
        ...
    def __init__(self: Test, k: int):
        ...
```

Allowing defaults with `__init__` is not straightforward since we also need to allow overloaded `__init__`, but it's something we could possibly add in the revised type-checker.

**Comment by inumanag** [comment]
Resolved in #132
